cli/src/main/java/com/devonfw/tools/ide/commandlet/ShellCommandlet.java

```diff
@@ -79,13 +81,13 @@ public void run() {
 
         // TODO: implement TailTipWidgets, see: https://github.com/devonfw/IDEasy/issues/169
 
-        String prompt = "ide> ";
         String rightPrompt = null;
         String line;
 
         AnsiConsole.systemInstall();
         while (true) {
           try {
+            String prompt = "ide " + context.getCwd() + "> ";
```

Comment:
I see some general problem here with the design of our shell feature:
* Initially we only supported IDEasy commandlets.
* Therefore we changed the promt to `ide>`
* In other words if submit `ide> «args»` in our shell, you will get more or less the same effect as if you call `ide «args»` outside the shell.
* Now we start integration other non `ide` commands to the shell like `cd` (how about `icd`, `find`, `mv`, `cp`, `grep`, `cat`, etc. ?) this does not make sense anymore
* Instead, I would suggest that we pre-define the user input to `"ide "`  as if the user would have typed this manually.
* Then the user can use our shell like before
* But he also has the possibility to use Backspace to remove this pre-defined input and change it to `cd some-folder`.

If you want to do a quick fix, you can do a KISS solution and switch:
```suggestion
            String prompt = context.getCwd() + "$ ide ";
```
But IMHO it is inconsistent that the user does `ide cd some-directory` while the is no `cd` commandlet.

Further, it is a great idea to show the current working directory like in other shells.
However, the CWD/PWD can get very long:
![image](https://github.com/user-attachments/assets/a8ca328d-0382-44da-9c64-711e3da6d481)
This is a common problem discussed about Linux forums
Simple solution is to only print the file-name of the CWD (foldername instead of full path).
Or to limit the length - see e.g.
https://superuser.com/questions/1346795/multi-line-bash-prompt-if-path-is-longer-than-x-characters
We should create a separate ticket for this last thought, but it would be good to also address this.

My suggestion would be that you create two issues (one for the pre-define the user "ide " and one for the handling of long path to avoid line wrapping) and then just apply some pragmatic solution like my proposed change that already prepares for the first story. Then we merge this PR and other developers can take over implementing the new stories.

Reply:
Thanks for creating the issues:

* #821
* #822

👍